code-collab-app/src/homepage.js

```diff
@@ -13,35 +13,16 @@ class Homepage extends React.Component {
   }
 
   handleJoinSession(session_info) {
+    this.props.peer_data.reset();
     this.props.peer_data.join_session(session_info.document_name, session_info.id);
-    // If only peer in the session, get doc data from tracker
-    if(session_info.peers.length === 0) {
-      this.props.peer_data.get_document_data(session_info.id);
-      this.props.peer_data.socket.on('rcv_doc_data', (data) => {
-        this.props.peer_data.doc_data = data.doc;
-        this.props.history.push({
-          pathname: '/editor',
-          data: this.props.peer_data.doc_data,
```

Comment:
Not passing any data here because peer_data is directly injected into editor component now.

Reply:
Ok sounds good. But just one thing, since these socket events are asynchronous, have we tested to make sure the doc_data that the editor page is receiving is from the tracker and not the default (empty array)? I had this here to make sure we wait for the tracker's data before we send data to the editor page so I just want to make sure this is still being handled.

Reply:
It is receiving the doc_data from the peer_data store as a prop, so any changes made by sockets will update that and it will be pushed to Editor page. So I believe it is all good, I will test some more to make sure.

Reply:
How are we guaranteeing that the editor page will have the correct document data when we load it since we aren't passing it any data? (i.e., when does the peer request the document data from the other peers or tracker? before or after loading the page? how does the peer know what document to retrieve?)

Reply:
Each session has one document, and we do know which session we are joining. In case of asking the peers, all the peers in the session will have the same document so we will get the right one. And in case of the tracker, we do provide the session id (peer_data has access to session id).